execution_engine/src/core/engine_state/era_validators.rs

```diff
@@ -0,0 +1,47 @@
+use casper_types::{
+    auction::{EraId, ValidatorWeights},
+    ProtocolVersion,
+};
+
+use crate::shared::newtypes::Blake2bHash;
+
+#[derive(Debug)]
+pub enum GetEraValidatorsResult {
+    /// A successful query operation returned validator weights for given era.
+    Success { validator_weights: ValidatorWeights },
```

Comment:
`ValidatorWeights` is a map from public keys to `U512`s - would it be possible to make the value type `Motes` instead? The weights are actually amounts of motes staked, so it would make sense semantically, and it would save me a conversion on the consensus side.

Reply:
Agreed.

Reply:
On the other hand you could argue that consensus shouldn't have to care about how the weights are determined: If instead of PoS, we had some complicated on-chain governance process that has nothing to do with stakes, it should work just the same way. So I'd prefer if consensus actually didn't receive an amount in `Motes`; as a general rule, I think the whole consensus component shouldn't _import_ types like `Motes`.
(Not important, of course, and not a blocker.)

Reply:
That makes sense, too. We are now initializing consensus based on values in `Motes`, because that's what we get from the chainspec, but we could in principle only care about `U512` values and convert the `Motes` somewhere external to consensus.